src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -22687,6 +22687,118 @@ .locals init (<>y__InlineArray1<int> V_0)
+                //     [CollectionBuilder(typeof(ImmutableArray), "Create")]
+                Diagnostic(ErrorCode.WRN_SameFullNameThisAggAgg, "ImmutableArray").WithArguments("0.cs", "System.Collections.Immutable.ImmutableArray", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable.ImmutableArray").WithLocation(7, 31),
+                // 0.cs(14,23): warning CS0436: The type 'ImmutableArray<T>' in '0.cs' conflicts with the imported type 'ImmutableArray<T>' in 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in '0.cs'.
+                //         public static ImmutableArray<T> Create<T>(ReadOnlySpan<T> items) => default;
+                Diagnostic(ErrorCode.WRN_SameFullNameThisAggAgg, "ImmutableArray<T>").WithArguments("0.cs", "System.Collections.Immutable.ImmutableArray<T>", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable.ImmutableArray<T>").WithLocation(14, 23),
+                // 0.cs(21,23): warning CS0436: The type 'ImmutableArray<T>' in '0.cs' conflicts with the imported type 'ImmutableArray<T>' in 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in '0.cs'.
+                //         public static ImmutableArray<T> AsImmutableArray<T>(T[] array) => default;
+                Diagnostic(ErrorCode.WRN_SameFullNameThisAggAgg, "ImmutableArray<T>").WithArguments("0.cs", "System.Collections.Immutable.ImmutableArray<T>", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable.ImmutableArray<T>").WithLocation(21, 23),
+                // 1.cs(4,12): warning CS0436: The type 'ImmutableArray<T>' in '0.cs' conflicts with the imported type 'ImmutableArray<T>' in 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in '0.cs'.
+                //     static ImmutableArray<int> F() => [1, 2, 3];
+                Diagnostic(ErrorCode.WRN_SameFullNameThisAggAgg, "ImmutableArray<int>").WithArguments("0.cs", "System.Collections.Immutable.ImmutableArray<T>", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable.ImmutableArray<T>").WithLocation(4, 12),
+                // 1.cs(4,39): error CS9203: A collection expression of type 'ImmutableArray<int>' cannot be used in this context because it may be exposed outside of the current scope.
+                //     static ImmutableArray<int> F() => [1, 2, 3];
+                Diagnostic(ErrorCode.ERR_CollectionExpressionEscape, "[1, 2, 3]").WithArguments("System.Collections.Immutable.ImmutableArray<int>").WithLocation(4, 39));
```

Comment:
I'm confused why this is illegal. The value `[1, 2, 3]` converts to `ReadOnlySpan<int>` here but that is returnable because it's using the RVA optimization. Effectively this is legal

```csharp
ReadOnlySpan<int> M() {
  ReadOnlySpan<int> span = [1, 2, 3];
  return span;
}
```

Given that I would expect that the result of `ImmutableArray.Create([1, 2, 3])` also be returnable.

Reply:
I think if the ImmutableArray case in HasLocalScope were grouped with the CollectionBuilder case, then the expected behavior would probably happen.

FWIW, I also feel completely fine with saying: you defined ImmutableArray as a ref struct? Ok, bad inconvenient things will happen, although the compiler won't outright crash.

Reply:
`CollectionBuilderMethod` is only set for `CollectionExpressionTypeKind.CollectionBuilder` case so if we group `ImmutableArray` with that case in `HasLocalScope`, we'll consider the builder method as invalid and treat the expression as local scope always.